libvirt/resource_libvirt_network.go

```diff
@@ -0,0 +1,313 @@
+package libvirt
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	libvirt "github.com/dmacvicar/libvirt-go"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"net"
+)
+
+// a libvirt network resource
+//
+// Resource example:
+//
+// resource "libvirt_network" "k8snet" {
+//    name = "k8snet"
+//    domain = "k8s.local"
+//    mode = "nat"
+//    addresses = ["10.17.3.0/24"]
+// }
+//
+// "addresses" can contain (0 or 1) ipv4 and (0 or 1) ipv6 ranges
+// "mode" can be one of: "nat" (default), "isolated"
+//
+func resourceLibvirtNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLibvirtNetworkCreate,
+		Read:   resourceLibvirtNetworkRead,
+		Update: resourceLibvirtNetworkUpdate,
+		Delete: resourceLibvirtNetworkDelete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"domain": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"mode": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"bridge": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+			},
+			"addresses": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Required: false,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```

Comment:
Maybe we should use `ip_range` instead of `addresses`, because some other providers call it that way...